glotaran/parameter/parameter_group.py

```diff
@@ -69,7 +70,7 @@ def __init__(self, label: str = None, root_group: ParameterGroup = None):
     @classmethod
     def from_dict(
         cls,
-        parameter_dict: dict[str, dict | list],
+        parameter_dict: dict[str, dict[str, Any] | list[float | list[Any]]],
```

Comment:
How to type this properly.

Reply:
This is something we should do in our new `glotaran.typing` package and will need thinking due to the nested structure of `ParameterGroup` and all the different ways to represent a `Parameter`. Most likely some kind of `Mapping` + `Protocol` mix.